Source/Core/VideoCommon/ShaderGenCommon.h

```diff
@@ -218,57 +220,34 @@ void WriteSwitch(ShaderCode& out, APIType ApiType, std::string_view variable,
                  const Common::EnumMap<std::string_view, last_member>& values, int indent,
                  bool break_)
 {
-  const bool make_switch = (ApiType == APIType::D3D);
-
   // The second template argument is needed to avoid compile errors from ambiguity with multiple
   // enums with the same number of members in GCC prior to 8.  See https://godbolt.org/z/xcKaW1seW
   // and https://godbolt.org/z/hz7Yqq1P5
   using enum_type = decltype(last_member);
 
-  // {:{}} is used to indent by formatting an empty string with a variable width
-  if (make_switch)
```

Comment:
Do we know why switch statements were avoided on OGL/Vk?  We have switch statements elsewhere, and I had no trouble with glslang's handling of switch statements here when working on the Metal renderer, and I think they look much nicer than the alternative massive if/else mess.

Reply:
Our Opengl backend is defaulted to a very old version.  From what I can tell, Android as well didn't have a a switch statement in ES 2.0.  But those are just guesses, we'd have to ask the original author to get a definitive answer ( @Pokechu22 ).

Also probably goes without saying but this PR isn't meant to change any glsl code, only derive HLSL from the existing content.  So any changes we want to do would need to be in a separate PR.

Reply:
I wasn't the one that chose to avoid switches; `UberShaderPixel` was previously written like that before I added the function (see f53dc6564fe16c23234b28907003d0d4cb1c4c5c - though I should have copied the comment from there when I added `WriteSwitch`).  The if tree was done by @stenzek in #5880.

I'm not sure if this is still a useful optimization or not.  Maybe drivers have been improved, or maybe they haven't.